File: src/main/java/org/minturtle/careersupport/user/repository/UserRepository.java
package org.minturtle.careersupport.user.repository;


import org.minturtle.careersupport.common.aop.Logging;
import org.minturtle.careersupport.user.entity.User;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

@Logging
public interface UserRepository extends ReactiveMongoRepository<User, String> {
    Mono<User> findByUsername(String username);
}

File: src/main/kotlin/org/minturtle/careersupport/user/repository/UserRepository.kt
package org.minturtle.careersupport.user.repository

import org.minturtle.careersupport.user.entity.User
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface UserRepository : CoroutineCrudRepository<User, String> {

    suspend fun findByUsername(username : String) : User?
}